### ISSUE TITLE
NimBLE: ble_svc_gap.c fails to compile when build configured to detect uninitialized variable (IDFGH-1335)

### ISSUE THREAD

**Issue by bfriedkin** [espressif/esp-idf#3623]
Hello -

I get the following error when my build is configured to generate errors on uninitialized variables:

```
/Users/brianfriedkin/esp32/esp-idf/components/nimble/nimble/nimble/host/services/gap/src/ble_svc_gap.c: In function 'ble_svc_gap_access':
/Users/brianfriedkin/esp32/esp-idf/components/nimble/nimble/nimble/host/services/gap/src/ble_svc_gap.c:195:9: error: 'rc' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     int rc;
```

Initializing `rc` to `0` resolves the problem, though I don't know if that's a preferred solution.

Regards,
Brian

**Comment by dhrishi** [comment]
@bfriedkin Thanks. I will fix this

**Comment by dhrishi** [comment]
This has been fixed with commit 416e753b7c3e539c44e0fb743f22c75bff7a42b6
